include/boost/unordered/detail/foa/table.hpp

```diff
@@ -313,6 +328,9 @@ class table:table_core_impl<TypePolicy,Hash,Pred,Allocator>
   using locator=typename super::locator;
   using compatible_concurrent_table=
     concurrent_table<TypePolicy,Hash,Pred,Allocator>;
+  using group_type_pointer=typename boost::pointer_traits<
+    typename boost::allocator_pointer<Allocator>::type
+  >::template rebind<group_type>;
```

Comment:
Not sure I understand. The only mentions of GroupPtr are in the `table_iterator`s where they belong. Here we're rebinding because we have the Allocator whose `value_type` is not group_type.

For example, this static assertion fails:
```cpp
static_assert(std::is_same_v<
  typename boost::pointer_traits<boost::allocator_pointer_t<Allocator>>::element_type,
  group_type>);
```
with:
```
../../boost/unordered/detail/foa/table.hpp:331:3: error: static assertion failed due to requirement 'std::is_same_v<std::pair<const x::D, x::D>, boost::unordered::detail::foa::group15<boost::unordered::detail::foa::plain_integral>>'
  static_assert(std::is_same_v<typename boost::pointer_traits<boost::allocator_pointer_t<Allocator>>::element_type,group_type>);
```

Reply:
Please ignore, I badly misread the code, everything's fine on your side.